pkg/catalogsourceconfig/configuring.go

```diff
@@ -148,45 +124,20 @@ func (r *configuringReconciler) reconcileCatalogSource(csc *v1alpha1.CatalogSour
 	return nil
 }
 
-// reconcileConfigMap ensures a ConfigMap exists with all the Operator artifacts
-// in its Data section
-func (r *configuringReconciler) reconcileConfigMap(csc *v1alpha1.CatalogSourceConfig) error {
-	// Construct the operator artifact data to be placed in the ConfigMap data
-	// section.
-	data, err := r.createCatalogData(csc)
-	if err != nil {
-		return err
-	}
-
-	// Check if the ConfigMap already exists
-	configMapName := csc.Name
-	configMapGet := new(ConfigMapBuilder).WithTypeMeta().ConfigMap()
-	key := client.ObjectKey{
-		Name:      configMapName,
-		Namespace: csc.Spec.TargetNamespace,
+// checkPackages returns an error if there no valid packages present in the
+// datastore.
+func (r *configuringReconciler) checkPackages(csc *v1alpha1.CatalogSourceConfig) error {
+	atLeastOneFound := false
+	packageIDs := GetPackageIDs(csc.Spec.Packages)
+	for _, packageID := range packageIDs {
+		if _, err := r.reader.Read(packageID); err != nil {
+			continue
```

Comment:
Suggestion: It would be useful to log the error(s) we run into. `NewAggregate` function from   `k8s.io/apimachinery/pkg/util/errors` can be used to return aggregated error(s), if that works for you.